Add docs to better explain what goes into a PR

milch opened the pull request.

milch committed:
Add docs to better explain what goes into a PR

milch committed:
Add docs to better explain what goes into a PR

milch committed:
Fix doubly empty lines

milch committed:
Fix doubly empty lines

KrauseFx commented:
Beautiful 👍

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+
+## Finding things to work on
+
+The core team usually tags issues that are ready to be worked on and easily accessible for new contributors with the [“you can do this” label][you can do this]. If you’ve never contributed to _fastlane_ before, these are a great start!
```
"these are a great start" -> "these are a great place to start"

mfurtak commented on YourFirstPR.md:
````diff
@@ -0,0 +1,77 @@
+bundle exec rubocop -a
+```
+
+After introducing some changes to the _fastlane_ source code, you probably want to test the changes for your application. The easiest way to do so it use [bundler](https://bundler.io/).
````
"bundler" the project name is usually capitalized "Bundler"

mfurtak replied:
The `bundle` command has already been used in examples a number of times before this. Does it make sense to move this statement higher in the document, and explain super briefly what it does?

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+
+After introducing some changes to the _fastlane_ source code, you probably want to test the changes for your application. The easiest way to do so it use [bundler](https://bundler.io/).
+
+Copy the Gemfile [.assets/Gemfile](.assets/Gemfile) from your local fastlane clone and drop it into your project's root folder.
```
Now that all sub-projects refer to projects' source code (instead of Gems) by default, is this instruction still necessary?

mfurtak replied:
fastlane -> `_fastlane_`

milch replied:
> Now that all sub-projects refer to projects' source code (instead of Gems) by default, is this instruction still necessary?

By "your project's root folder" we are talking about the user's test project, where they want to test out there changes to fastlane.

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+
+## After the review
+
+Once a core member has reviewed your PR, you might need to make changes before it gets merged. To make it easier on us, please make sure to avoid using `git commit --amend` and force pushes so that only the changes you did after the review will be visible on GitHub. Feel free to use whatever commit messages you like, as we will squash them anyway. When you are done addressing your review, also ad ...
```
Suggestion:

"To make it easier on us... visible on GitHub." ->

"To make it easier on us, please make sure to avoid using `git commit --amend` or force pushes to make corrections. By avoiding rewriting the commit history, you will allow each round of edits to become its own visible commit. This helps the people who need to review your code easily understand exactly what has changed since the last time they looked."

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+
+## Checking out the fastlane repo
+
+- Click the “fork” button in the upper right corner of the [main fastlane repo][fastlane]
```
Nitpick: "Fork" button 😄

mfurtak replied:
fastlane -> `_fastlane_`

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+
+If you want to work on something else, e.g. new functionality or fixing a bug, we kindly ask you to submit a new issue, so that we can have a chance to discuss it first. We might have some pointers for you on how to get started, or how to best integrate it with existing solutions.
+
+## Checking out the fastlane repo
```
fastlane -> `_fastlane_`

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+- Create a new branch to work on:
+  - `git checkout -b <YOUR_BRANCH_NAME>`
+  - A good name for a branch describes the thing you’ll be working on, e.g. `docs-fixes`, `fix-deliver-upload`, `gym-build-android-app`, etc.
+- That’s it! Now you’re ready to work on fastlane 
```
fastlane -> `_fastlane_`

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+
+Copy the Gemfile [.assets/Gemfile](.assets/Gemfile) from your local fastlane clone and drop it into your project's root folder.
+
+Make sure to change the `local_fastlane_path` variable to point to your fastlane clone, e.g. `~/fastlane`, then you can run
```
fastlane -> `_fastlane_`

mfurtak commented on YourFirstPR.md:
```diff
@@ -0,0 +1,77 @@
+
+## Submitting the PR
+
+When the coding is done and you’re finished testing your changes, you are ready to submit the PR to the [fastlane main repo][fastlane]. Everything you need to know about submitting the PR itself is inside our [Pull Request Template][pr template]. Some best practices are:
```
fastlane -> `_fastlane_`

mfurtak commented:
Looking good! I left some suggestions, questions, and small formatting corrections 👍

Commit:
Address review

Commit:
Address review

milch commented:
@mfurtak Thanks, I addressed your reviews!

mfurtak commented:
🙌  💯  Do it!

mfurtak committed:
Fix xcodebuild action specs (#7406)

* Fix xcodebuild action specs

As of 2016-12-7, CircleCI has started setting the FL_BUILDLOG_PATH
environment variable, which changes the default output location for
build logs.

Many specs were assuming that the generated commands
would include the default log path, rather than using the buildlog_path
helper function that takes into account the environment variable.

This fixes those specs to do the right thing.

* Adjust the FL_BUILDLOG_PATH on CircleCI to be the original expected location

milch committed:
Change CONTRIBUTING.md to act as a 'table of contents' for the new contributor docs (#7393)

* Change contributing to act as a 'table of contents' for the new contributor docs

* Surround all fastlanes with _

* So long and thanks for all the fish

KrauseFx commented:
Just waiting for the build 👍

Commit:
Merge branch 'submit-pr-docs' of https://github.com/fastlane/fastlane into submit-pr-docs
